src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2742,6 +2742,39 @@ private function registerMailerConfiguration(array $config, ContainerBuilder $co
             $container->removeDefinition('mailer.messenger_transport_listener');
         }
 
+        if ($config['dkim_signer']['enabled']) {
```

Comment:
There case where it is enabled should check whether the listener class exists, throwing an exception otherwise to tell the dev that they must update the `symfony/mailer` component to be able to enable those features.